liferay-frontend/liferay-portal
# LPS-116361 Remote web apps scaffolding and PoC

Please see commit messages for a description of the functionalities contained in the PR. I suggest starting the review by looking 
at each commit first, instead of going to the Files changed section directly.

---

The way to test this is:

1. Start the portal (maybe deleting the DB first if the remotewebappentry table is not created on the first try).
2. Edit the home page (click the pencil)
3. In Fragments and Widgets click on the Widgets tab
4. Find the SAMPLE category
5. Drag and drop entry El País to the page

That should create a portlet which renders an iframe pointing to http://www.elpais.com.

---

To test the Remote Web App admin portlet:

1. Open the Applications Menu
2. Select `Application` tab
3. Click on `Remote Web Apps` in the `CUSTOM APPS` section

The portlet to search and edit remote web apps should appear.

---

:warning: :warning: :warning: 
Currently, the persistence service is not being injected in `RemoteWebAppEntryLocalServiceImpl` for an unknown reason. My guess is that service builder or project configuration is lacking something, but as I don't have any experience with service builder and I haven't had enough time to investigate it I've been unable to fix it :disappointed:
